### DISCUSSION TITLE
Can't login to Tools->Settings, but can to ALL the other Tools->

### DISCUSSION THREAD

**Discussion by Astrobirder** [mcguirepr89/BirdNET-Pi#609] [Q&A]
Subject line says most of it.
This is a NEW install, and my first installation.
I get the pop-up for logging in when I first choose Tools->
I'm able to login with birdnet  and no password, and I'm able to choose ALL of the Tools EXCEPT Settings, and they operate as expected.

When I choose SETTINGS, I get another prompt for Userid and Password and neither birdnet with no password, nor my ssh userid/password work.  When I cancel out, I get the page "Basic Settings" with "You cannot edit the settings for this installation".

Ideas?  I was guessing maybe permissions on a file that stores the Basic Settings, but I'm not sure where that is--maybe birdnet.conf?  Currently that file is UID= pi   GID= pi  with permissions of 664.

Thanks!

**Comment by ehpersonal38** [comment]
Hi there!

My best guess is that the settings page PHP script is having some kind of error, which results in the authentication prompt being displayed over and over. 

Before we look deeper into this, is the latitude and longitude of your birdnet.conf filled out properly? I think I recall this happening to someone whose lat/lon were blank. 

If you could paste your whole birdnet.conf with any sensitive bits redacted I can verify that it all looks good there.

**Comment by owai1980** [comment]
Be careful with the coma in the gps coorfinates.

, or.

Not the same!

Le mar. 18 oct. 2022 à 02:18, ehpersonal38 ***@***.***> a
écrit :

**Comment by Astrobirder** [comment]
Well, it doesn't appear to be the comma in the GPS coordinates.  Maybe it needs E,W,N,S instead of positive/negative numbers?  Attached is the whole file.  I replaced passwords (with Bs) and GPS degree coordinates got changed to ##, but other than that it is complete.  I also did a cat -vet to look for unprintable characters (an old sysadmin trick).  

I know that script languages don't take too kindly to strange characters in config files :-). I didn't see anything odd.  I also went looking to find the WEBROOT and any logs and didn't find them quickly.  Is there somewhere that any of the PHP script errors would be logged?  A quick lsof -i -P -n | grep LISTEN didn't show a webserver running, but clearly there is one--perhaps its only spawning when someone knocks on port 80?

Oh, and I had to add a .txt to the end of the filename to attach here.
[birdnet.conf.txt](https://github.com/mcguirepr89/BirdNET-Pi/files/9824966/birdnet.conf.txt)

**Reply by ehpersonal38** [reply]
Your birdnet.conf looks good, so yeah we're going to need to get some logs out of PHP to further diagnose the issue.

I believe the way I enabled error logging on my machine was to edit the file `/etc/php/7.4/fpm/php.ini` and set `error_log = syslog`. You might need to set `log_errors = On`, too. Then you can `sudo service php7.4-fpm restart`. Then you should see PHP errors written to the end of `/var/log/syslog`. If this doesn't work you may need to repeat the above steps on `/etc/php/7.4/fpm/php-fpm.conf`, too.

**Reply by Astrobirder** [reply]
Wow--thanks for the quick response.  I'll do that now!  Also, when I open the Spectrogram tab, I get a perpetual Loading...   and no spectrogram.

**Reply by Astrobirder** [reply]
While watching tail -f /var/log/syslog, I notice this when I tried clicking on Settings:
Oct 19 20:28:56 birdnetpi php: PHP Warning:  syntax error, unexpected '!' in ./scripts/thisrun.txt on line 4
Oct 19 20:28:56 birdnetpi php:  in /home/pi/BirdNET-Pi/homepage/views.php on line 19

Hmm, it appears that maybe the php code doesn't like ! inside a password?
I have one in my current CADDY_PWD.

**Reply by Astrobirder** [reply]
Yep, that's totally the problem.  All kinds of things now work much better! The Spectrogram page even appears to load now.
A very strange bug!

Now to figure out why my new working AOM5024 microphone isn't working with the Ugreen black sound card!

**Reply by ehpersonal38** [reply]
Glad that helped and great to see that you got to the bottom of the issue! Special characters not working in passwords is a known bug https://github.com/mcguirepr89/BirdNET-Pi/issues/276, but Patrick has been working on a complete overhaul of BirdNET-Pi which will solve that bug and many more, so it's not a huge priority for us right now on this version.

> Now to figure out why my new working AOM5024 microphone isn't working with the Ugreen black sound card!

I'm not an expert in this audio stuff at all, but here's some reasons why my AOM5024 hasn't worked in the past. You probably already know them but a sanity check never hurts :)
1. Your microphone audio jack is incompatible with your UGreen sound card (TRS vs TRRS)
2. Your capsule is soldered incorrectly
3. Your microphone gain is 0 https://github.com/mcguirepr89/BirdNET-Pi/wiki/Adjusting-your-sound-card

Take care!

**Reply by Astrobirder** [reply]
Well, I followed @DD4WH 's #39 directions, with the exception of trying to shield the short wires at the capsule end (so as to make sure there is no shield/foil touching the capsule, as I didn't have the correct heat shrink to pre-wrap the capsule's grounded casing).

I tested the wired microphone with Audacity and a laptop, and it recorded the sounds I made.  

I did do your suggestion #3, and it didn't seem to change anything.  I know the Pi is working, because I've hooked up a Logitech wireless headset and used it's microphone and did get bird recognitions while playing back from the Sibley app on my phone.  So it's back to the Ugreen soundcard and potentially the pi choosing it properly.

I did go back and look at syslog (as I still have the php logging setup going) and I'm seeing this scrolling continuously:

Oct 20 14:25:34 birdnetpi livestream.sh[1158]: [mp3 @ 0x559d0dba00] Application provided invalid, non monotonically increasing dts to muxer in stream 0: 3618613844 >= 3063866769

So it seems maybe something with the Pulseaudio / ALSA setup?

Forgot to mention that I tried with BOTH the White and the BLACK Ugreen soundcards that were suggested.  I have a TRS plug on the microphone, so I think that's correct.

**Reply by ehpersonal38** [reply]
> So it seems maybe something with the Pulseaudio / ALSA setup?

I agree that the issue probably lies here. So just to clarify, when you run the `alsamixer -m` command in your terminal as per my above #3 suggestion, and then press `F4` to go into capture mode then `F6` to select a sound card, does the card show up as an option or not? Here's how my working mic shows up:

![chrome_VLt1XsEIgm](https://user-images.githubusercontent.com/103586016/197046126-efa0d623-2f42-4056-9b16-f3fa621eed10.png)

**Reply by Astrobirder** [reply]
Yes, that's basically how mine appears:
![image](https://user-images.githubusercontent.com/13646631/197064426-c0885e33-d327-4dc3-b396-3216eefac156.png)

**Reply by ehpersonal38** [reply]
Okay that's super strange then... I'm not even sure what it could be. You're definitely welcome to open a new discussion and hopefully some audio wizards will chime in with suggestions.

**Reply by Astrobirder** [reply]
Probably a good idea.  I've always been of the opinion that Pulseaudio is evil :-).   Mostly because I never have really understood it.
